crates/duty-tracker/src/contract_manager.rs

```diff
@@ -155,36 +214,37 @@ impl ContractManagerCtx {
                         .commit(deposit_txid, contract.state())
                         .await?;
                     if let Some(duty) = duty {
-                        todo!() // TODO(proofofkeags): execute duty
+                        duties.push(duty);
                     }
                 }
             }
         }
 
         // Now that we've handled all the transaction level events, we should inform all the
         // CSMs that a new block has arrived
-        for (deposit_txid, contract) in self.active_contracts.iter_mut() {
+        for (_, contract) in self.active_contracts.iter_mut() {
             if let Some(duty) = contract.process_contract_event(ContractEvent::Block(height))? {
-                todo!() // TODO(proofofkeags): execute duty
+                duties.push(duty);
             }
         }
 
+        for duty in duties {
+            self.execute_duty(duty);
+        }
+
         Ok(())
     }
 
     async fn process_p2p_message(&mut self, msg: GossipsubMsg) -> Result<(), ContractManagerErr> {
         match msg.unsigned {
-            UnsignedGossipsubMsg::StakeChainExchange {
-                stake_chain_id,
-                info,
-            } => todo!(),
+            UnsignedGossipsubMsg::StakeChainExchange { .. } => todo!(),
```

Comment:
And this will change in https://github.com/alpenlabs/strata-p2p/pull/37 to https://github.com/alpenlabs/strata-p2p/blob/82ccb672589d2a2ea8e7dbb42e0a8bfe893c8b33/crates/wire/src/p2p/v1/typed.rs#L219-L230:

```rs
    /// Operators exchange stake chain info.
    StakeChainExchange {
        /// 32-byte hash of some unique to stake chain data.
        stake_chain_id: StakeChainId,

        /// [`Txid`] of the pre-stake transaction.
        pre_stake_txid: Txid,

        /// vout of the pre-stake transaction.
        pre_stake_vout: u32,
    },
```